### DISCUSSION TITLE
How to setup characters config and phonemes for bangla?

### DISCUSSION THREAD

**Discussion by mobassir94** [coqui-ai/TTS#1996] [General Q&A]
i see in most tutorials people are using characters for english,german etc latin like languages like this :
```
 characters={
        "pad": "_",
        "eos": "~",
        "bos": "^",
        "characters": "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyzŋ!'(),-.:;? ",
        "punctuations": "!'(),-.:;? ",
        "phonemes": "iy\u0268\u0289\u026fu\u026a\u028f\u028ae\u00f8\u0258\u0259\u0275\u0264o\u025b\u0153\u025c\u025e\u028c\u0254\u00e6\u0250a\u0276\u0251\u0252\u1d7b\u0298\u0253\u01c0\u0257\u01c3\u0284\u01c2\u0260\u01c1\u029bpbtd\u0288\u0256c\u025fk\u0261q\u0262\u0294\u0274\u014b\u0272\u0273n\u0271m\u0299r\u0280\u2c71\u027e\u027d\u0278\u03b2fv\u03b8\u00f0sz\u0283\u0292\u0282\u0290\u00e7\u029dx\u0263\u03c7\u0281\u0127\u0295h\u0266\u026c\u026e\u028b\u0279\u027bj\u0270l\u026d\u028e\u029f\u02c8\u02cc\u02d0\u02d1\u028dw\u0265\u029c\u02a2\u02a1\u0255\u0291\u027a\u0267\u02b2\u025a\u02de\u026b",
        "unique": True
    },
```

however i would like to do something similar for non-latin language bangla/bengali.

i was trying something like this instead

```
characters_config = CharactersConfig(
pad = None,
eos = '।',
bos = None,
phonemes = None,
characters = "হথশ৫কওয০গদড়খয়ঋনঅ৪এবঠঢ৭৯ধঙটঝৎণতর২চঌড৬ঔপভমঢ়ঈ৮ঘ১ষ৩ফছলজআ।ঊইসঐউঞ া ্ ু ী ে ং ি ় ঁ ৃ ো ূ ৈ ৌ ঃ"
)
```
your library by default doesn't support bangla,so with [espeak-ng](https://github.com/lingjzhu/CharsiuG2P/tree/main/sources/espeak-ng) installed,inside GlowTTSConfig class i set use_phonemes=True,phoneme_language="bn",
for bangla language what pad,bos ane specially phonemes values i need to use? i am not able to figure it out yet,can anyone expert in this TTS domain help please? @erogol

**Comment by AIFahim** [comment]
Hello Brother. Hope you are doing great. Did you able to train for bangla?

**Comment by mobassir94** [comment]
Hi brother @AIFahim 
yeah i was able to train bangla tts, please check this code that i used for training bangla tts -> https://github.com/mobassir94/comprehensive-bangla-tts/blob/main/ben_glow_tts/train_bn_glow_tts_male.ipynb
thanks

**Reply by AIFahim** [reply]
Thank you so much, brother. I found you didn't use phonemes for the training. Is there any way to use the phonemes for this project for bangla?

**Reply by sudip550** [reply]
Can you please tell me that how to train voice model in hindi language. Please help sir

**Comment by mobassir94** [comment]
@AIFahim i didn't use phonemizer because gruut doesn't support bangla and i couldn't put enough trust on espeak. However espeak has support for bangla https://github.com/espeak-ng/espeak-ng/blob/master/docs/languages.md and if you want to use bangla phonemes you can use espeak wrapper as shown here https://github.com/coqui-ai/TTS/blob/dev/TTS/tts/utils/text/phonemizers/espeak_wrapper.py or make your own bangla G2P model that is robust enough like gruut, good luck.

**Reply by sudip550** [reply]
Please tell sir, how to train voice model in hindi

**Comment by AIFahim** [comment]
Brother can you help to share the if you know that which files send to the tensorboard while training or valdation?

**Comment by mobassir94** [comment]
sorry, i didn't use tensorboard

**Comment by erogol** [comment]
If you are using phonemes then you should just use the phoneme characters of the phoneme backend. For instance, if you are using ESpeak probably it is enough to use the default phoneme set. 

If you are training with normal characters then disable phoneme extraction and enter characters into the config as you've done above.

**Comment by adil-ahmed** [comment]
Hello @mobassir94 and @erogol 

I was also trying to train a bangla dataset using VITS model. Followed similar approach but somehow stuck. Could you please share where i am doing wrong. 

my character parameter looks like this:

`
characters = CharactersConfig(
pad="<PAD>",
eos="<|>",
bos="<BOS>",
blank="<BLNK>",
phonemes=None,
characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789ঁংঃঅআইঈউঊঋএঐওঔকখগঘঙচছজঝঞটঠডঢণতথদধনপফবভমযরলশষসহ়ািীুূৃেৈোৌ্ৎড়য়০১২৩৪৫৬৭৮৯ৰ",
punctuations = "“-–‘’”!\"#%'()*,./:;?[]°। "
)
`


This is the error i am getting:

`> Setting up Audio Processor...
 | > sample_rate:22050
 | > resample:False
 | > num_mels:80
 | > log_func:np.log10
 | > min_level_db:0
 | > frame_shift_ms:None
 | > frame_length_ms:None
 | > ref_level_db:None
 | > fft_size:1024
 | > power:None
 | > preemphasis:0.0
 | > griffin_lim_iters:None
 | > signal_norm:None
 | > symmetric_norm:None
 | > mel_fmin:0
 | > mel_fmax:None
 | > pitch_fmin:None
 | > pitch_fmax:None
 | > spec_gain:20.0
 | > stft_pad_mode:reflect
 | > max_norm:1.0
 | > clip_norm:True
 | > do_trim_silence:False
 | > trim_db:60
 | > do_sound_norm:False
 | > do_amp_to_db_linear:True
 | > do_amp_to_db_mel:True
 | > do_rms_norm:False
 | > db_level:None
 | > stats_path:None
 | > base:10
 | > hop_length:256
 | > win_length:1024
 | > Found 7390 files in /content/drive/MyDrive/All/Personal/UBishops/ModelTraining/AAD/VCTK_Bangla_22.05k/vctk-bn-16k-ds-flac
 > initialization of speaker-embedding layers.
fatal: not a git repository (or any parent up to mount point /content)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: not a git repository (or any parent up to mount point /content)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
 > Training Environment:
 | > Backend: Torch
 | > Mixed precision: True
 | > Precision: fp16
 | > Current device: 0
 | > Num. of GPUs: 1
 | > Num. of CPUs: 2
 | > Num. of Torch Threads: 1
 | > Torch seed: 54321
 | > Torch CUDNN: True
 | > Torch CUDNN deterministic: False
 | > Torch CUDNN benchmark: False
 | > Torch TF32 MatMul: False
2024-03-13 05:21:29.421497: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:9261] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-03-13 05:21:29.421543: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:607] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-03-13 05:21:29.422934: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1515] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2024-03-13 05:21:29.430264: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2024-03-13 05:21:30.477151: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
 > Start Tensorboard: tensorboard --logdir=/content/drive/MyDrive/All/Personal/UBishops/ModelTraining/AAD/OutputForBangla/vits_vctk_old_mini_gdrive_ft-4000-March-13-2024_05+21AM-0000000
 > `speakers.pth` is saved to /content/drive/MyDrive/All/Personal/UBishops/ModelTraining/AAD/OutputForBangla/vits_vctk_old_mini_gdrive_ft-4000-March-13-2024_05+21AM-0000000/speakers.pth.
 > `speakers_file` is updated in the config.json.

 > Model has 86419884 parameters

 > EPOCH: 0/1000
 --> /content/drive/MyDrive/All/Personal/UBishops/ModelTraining/AAD/OutputForBangla/vits_vctk_old_mini_gdrive_ft-4000-March-13-2024_05+21AM-0000000


> DataLoader initialization
| > Tokenizer:
	| > add_blank: True
	| > use_eos_bos: False
	| > use_phonemes: False
| > Number of instances : 7317
 | > Preprocessing samples
 | > Max text length: 200
 | > Min text length: 16
 | > Avg text length: 141.95741799578693
 | 
 | > Max audio length: 175424.0
 | > Min audio length: 25525.5
 | > Avg audio length: 109215.99729160397
 | > Num. instances discarded samples: 671
 | > Batch group size: 160.
/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py:557: UserWarning: This DataLoader will create 4 worker processes in total. Our suggested max number of worker in current system is 2, which is smaller than what this DataLoader is going to create. Please be aware that excessive worker creation might get DataLoader running slow or even freeze, lower the worker number to avoid potential slowness/freeze if necessary.
  warnings.warn(_create_warning_msg(

 > TRAINING (2024-03-13 05:21:35) 
সুতরাং যোগফল ১১*৫=৫৫
 [!] Character '=' not found in the vocabulary. Discarding it.
লক্ষ করি প্রথম ও শেষ পদের যোগফল ১+১০=১১
 [!] Character '+' not found in the vocabulary. Discarding it.
লক্ষ করি প্রথম ও শেষ পদের যোগফল ১+১০=১১
 [!] Character '=' not found in the vocabulary. Discarding it.
+৫+৬+৭+৮+৯+১০
 [!] Character '+' not found in the vocabulary. Discarding it.
ও ২৫ জন ছাত্র আছে দ্বিতীয় শ্রেণিতে মোট কত জন শিক্ষার্থী আছে বিয়োগ করি ৮৫ ১৩ = কত ২
 [!] Character '=' not found in the vocabulary. Discarding it.
৭ দিন = ১ সপ্তাহ ৩৬৫ দিন = ১বছর সময় বিষয়ক পাটিগনিত
 [!] Character '=' not found in the vocabulary. Discarding it.
x ৮ = ১৬ ২ x ৯ = ১৮ ২ ×১০ = ২০ ৩ এর গুণের নামতা ৩ x
 [!] Character '×' not found in the vocabulary. Discarding it.
৫ = ১৫ ৩ x ৬ = ১৮ ৩ × ৭ = ২১ ৩ x ৮ = ২৪
 [!] Character '×' not found in the vocabulary. Discarding it.
লক্ষ করলে দেখা যায় ১+১৯=২০ ৩+১৭=২০ ৫+১৫=২০ ইত্যাদি এরকম ৫ জোড়া সংখ্যা পাওয়া যায়
 [!] Character '+' not found in the vocabulary. Discarding it.
/usr/local/lib/python3.10/dist-packages/torch/functional.py:650: UserWarning: stft with return_complex=False is deprecated. In a future pytorch release, stft will return complex tensors for all inputs, and return_complex=False will raise an error.
Note: you can still call torch.view_as_real on the complex output to recover the old return format. (Triggered internally at ../aten/src/ATen/native/SpectralOps.cpp:863.)
  return _VF.stft(input, n_fft, hop_length, win_length, window,  # type: ignore[attr-defined]
৫× ১ = ৫ ৫× ২ = ১০ ৫× ৩ = ১৫ ৫× ৪
 [!] Character '×' not found in the vocabulary. Discarding it.

   --> TIME: 2024-03-13 05:21:58 -- STEP: 0/208 -- GLOBAL_STEP: 0
     | > loss_disc: 5.950425148010254  (5.950425148010254)
     | > loss_disc_real_0: 0.9786046743392944  (0.9786046743392944)
     | > loss_disc_real_1: 0.9720872044563293  (0.9720872044563293)
     | > loss_disc_real_2: 0.9970418214797974  (0.9970418214797974)
     | > loss_disc_real_3: 0.9804518818855286  (0.9804518818855286)
     | > loss_disc_real_4: 0.9917620420455933  (0.9917620420455933)
     | > loss_disc_real_5: 1.029815673828125  (1.029815673828125)
     | > loss_0: 5.950425148010254  (5.950425148010254)
     | > grad_norm_0: 0  (0)
     | > loss_gen: 5.949153900146484  (5.949153900146484)
     | > loss_kl: 173.41146850585938  (173.41146850585938)
     | > loss_feat: 0.28786540031433105  (0.28786540031433105)
     | > loss_mel: 74.85636138916016  (74.85636138916016)
     | > loss_duration: 1.607557773590088  (1.607557773590088)
     | > amp_scaler: 32768.0  (32768.0)
     | > loss_1: 256.1123962402344  (256.1123962402344)
     | > grad_norm_1: 0  (0)
     | > current_lr_0: 0.0002 
     | > current_lr_1: 0.0002 
     | > step_time: 5.7018  (5.701847314834595)
     | > loader_time: 17.2287  (17.22872543334961)

লক্ষ করি প্রথম ও শেষ পদের যোগফল ১+১০=১১
 [!] Character '+' not found in the vocabulary. Discarding it.
১৩=২^২+৩^২ ইত্যাদি সংখ্যা নিয়ে খেলা ১ দুই অঙ্কের যেকোনো সংখ্যা নাও সংখ্যার অঙ্ক দুইটির স্থান বদল করে
 [!] Character '^' not found in the vocabulary. Discarding it.
২=১^২+১^২ ৫=১^২+২^২ ৮=২^২+২^২ ১০=১^২+৩^২
 [!] Character '^' not found in the vocabulary. Discarding it.
৫=১^২+২^২ ৮=২^২+২^২
 [!] Character '^' not found in the vocabulary. Discarding it.
৫× ৮ = ৪০ ৫x ৯ = ৪৫ ৫×১০ = ৫০
 [!] Character '×' not found in the vocabulary. Discarding it.
১০=১^২+২^২ ১৩=২^২+৩^২ ইত্যাদি
 [!] Character '^' not found in the vocabulary. Discarding it.
যেমন সাধারণ তাপমাত্রায় হাইড্রোজেন ও অক্সিজেন গ্যাসীয় কিন্তু এদের থেকে উৎপন্ন যৌগ পানি সাধারণ তাপমাত্রায় তরল three point two পরমাণু ও অণু atoms and molecules
 [!] Character '\uf02d' not found in the vocabulary. Discarding it.

   --> TIME: 2024-03-13 05:24:51 -- STEP: 50/208 -- GLOBAL_STEP: 50
     | > loss_disc: 2.829683780670166  (3.177325577735901)
     | > loss_disc_real_0: 0.2484404444694519  (0.2942648637294769)
     | > loss_disc_real_1: 0.22439400851726532  (0.2979904474318027)
     | > loss_disc_real_2: 0.19146497547626495  (0.3003874598443507)
     | > loss_disc_real_3: 0.14283761382102966  (0.30504790216684335)
     | > loss_disc_real_4: 0.20850864052772522  (0.30077132821083064)
     | > loss_disc_real_5: 0.1726732701063156  (0.3156682564318181)
     | > loss_0: 2.829683780670166  (3.177325577735901)
     | > grad_norm_0: tensor(5.9951, device='cuda:0')  (tensor(2.7348, device='cuda:0'))
     | > loss_gen: 1.6249767541885376  (1.8057410955429076)
     | > loss_kl: 2.6901895999908447  (30.434691390991212)
     | > loss_feat: 0.8338497281074524  (0.45111045211553574)
     | > loss_mel: 37.465232849121094  (51.623901290893556)
     | > loss_duration: 2.1610050201416016  (1.906169831752777)
     | > amp_scaler: 1024.0  (1556.48)
     | > loss_1: 44.77525329589844  (86.22161415100098)
     | > grad_norm_1: tensor(155.1590, device='cuda:0')  (tensor(164.5702, device='cuda:0'))
     | > current_lr_0: 0.0002 
     | > current_lr_1: 0.0002 
     | > step_time: 2.2828  (1.8005088901519775)
     | > loader_time: 0.0579  (1.62012047290802)

যেমন সাধারণ তাপমাত্রায় হাইড্রোজেন ও অক্সিজেন গ্যাসীয় কিন্তু এদের থেকে উৎপন্ন যৌগ পানি সাধারণ তাপমাত্রায় তরল three point two পরমাণু ও অণু atoms and molecules
 [!] Character '\uf02d' not found in the vocabulary. Discarding it.
যেমন সাধারণ তাপমাত্রায় হাইড্রোজেন ও অক্সিজেন গ্যাসীয় কিন্তু এদের থেকে উৎপন্ন যৌগ পানি সাধারণ তাপমাত্রায় তরল three point two পরমাণু ও অণু atoms and molecules
 [!] Character '\uf02d' not found in the vocabulary. Discarding it.
যেমন সাধারণ তাপমাত্রায় হাইড্রোজেন ও অক্সিজেন গ্যাসীয় কিন্তু এদের থেকে উৎপন্ন যৌগ পানি সাধারণ তাপমাত্রায় তরল three point two পরমাণু ও অণু atoms and molecules
 [!] Character '\uf02d' not found in the vocabulary. Discarding it.
এটা দেশের গ্রাম বা মফস্বল শহরগুলোতে খুব বেশি সমস্যা না হলেও বড় বড় শহরকে ভাবিয়ে তুলেছে তিনি বলেন বাংলাদেশ সরকার ওয়াশ প্রোগ্রামের উন্নয়নে দৃঢ়প্রতিজ্ঞ
 [!] Character 'ঢ়' not found in the vocabulary. Discarding it.
অভাবজনিত রোগ ও প্রতিকার ভিটামিন a এর অভাবে রাতকানা রোগ হয় এর অভাব দীর্ঘস্থায়ী হলে চোখের কর্নিয়ায় আলসার হতে পারে— এ অবস্থাকে
 [!] Character '—' not found in the vocabulary. Discarding it.
এটা দেশের গ্রাম বা মফস্বল শহরগুলোতে খুব বেশি সমস্যা না হলেও বড় বড় শহরকে ভাবিয়ে তুলেছে তিনি বলেন বাংলাদেশ সরকার ওয়াশ প্রোগ্রামের উন্নয়নে দৃঢ়প্রতিজ্ঞ
 [!] Character 'ঢ়' not found in the vocabulary. Discarding it.

   --> TIME: 2024-03-13 05:28:24 -- STEP: 100/208 -- GLOBAL_STEP: 100
     | > loss_disc: 2.530229330062866  (2.939980375766754)
     | > loss_disc_real_0: 0.21263644099235535  (0.2576312400400637)
     | > loss_disc_real_1: 0.25169312953948975  (0.2769256188720464)
     | > loss_disc_real_2: 0.26189306378364563  (0.27834948889911165)
     | > loss_disc_real_3: 0.2771354913711548  (0.27824354305863386)
     | > loss_disc_real_4: 0.2864297926425934  (0.2768451030552388)
     | > loss_disc_real_5: 0.22813333570957184  (0.28206287972629085)
     | > loss_0: 2.530229330062866  (2.939980375766754)
     | > grad_norm_0: tensor(3.8269, device='cuda:0')  (tensor(4.8010, device='cuda:0'))
     | > loss_gen: 2.1431469917297363  (1.854775515794754)
     | > loss_kl: 2.1964669227600098  (16.393488152027125)
     | > loss_feat: 2.807081699371338  (1.0239125244319438)
     | > loss_mel: 34.33760452270508  (43.36080404281614)
     | > loss_duration: 2.2637391090393066  (2.0844331514835357)
     | > amp_scaler: 512.0  (1162.2400000000002)
     | > loss_1: 43.74803924560547  (64.71741355895993)
     | > grad_norm_1: tensor(137.7600, device='cuda:0')  (tensor(158.5187, device='cuda:0'))
     | > current_lr_0: 0.0002 
     | > current_lr_1: 0.0002 
     | > step_time: 3.1439  (1.976396610736847)
     | > loader_time: 5.3066  (1.8646897983551025)

এটা দেশের গ্রাম বা মফস্বল শহরগুলোতে খুব বেশি সমস্যা না হলেও বড় বড় শহরকে ভাবিয়ে তুলেছে তিনি বলেন বাংলাদেশ সরকার ওয়াশ প্রোগ্রামের উন্নয়নে দৃঢ়প্রতিজ্ঞ
 [!] Character 'ঢ়' not found in the vocabulary. Discarding it.
এটা দেশের গ্রাম বা মফস্বল শহরগুলোতে খুব বেশি সমস্যা না হলেও বড় বড় শহরকে ভাবিয়ে তুলেছে তিনি বলেন বাংলাদেশ সরকার ওয়াশ প্রোগ্রামের উন্নয়নে দৃঢ়প্রতিজ্ঞ
 [!] Character 'ঢ়' not found in the vocabulary. Discarding it.
অভাবজনিত রোগ ও প্রতিকার ভিটামিন a এর অভাবে রাতকানা রোগ হয় এর অভাব দীর্ঘস্থায়ী হলে চোখের কর্নিয়ায় আলসার হতে পারে— এ অবস্থাকে
 [!] Character '—' not found in the vocabulary. Discarding it.
অভাবজনিত রোগ ও প্রতিকার ভিটামিন a এর অভাবে রাতকানা রোগ হয় এর অভাব দীর্ঘস্থায়ী হলে চোখের কর্নিয়ায় আলসার হতে পারে— এ অবস্থাকে
 [!] Character '—' not found in the vocabulary. Discarding it.
অভাবজনিত রোগ ও প্রতিকার ভিটামিন a এর অভাবে রাতকানা রোগ হয় এর অভাব দীর্ঘস্থায়ী হলে চোখের কর্নিয়ায় আলসার হতে পারে— এ অবস্থাকে
 [!] Character '—' not found in the vocabulary. Discarding it.
যেমন− সাধারণ তাপমাত্রায় হাইড্রোজেন ও অক্সিজেন গ্যাসীয় কিন্তু এদের থেকে উৎপন্ন যৌগ পানি সাধারণ তাপমাত্রায় তরল three point two
 [!] Character '−' not found in the vocabulary. Discarding it.
যেমন− সাধারণ তাপমাত্রায় হাইড্রোজেন ও অক্সিজেন গ্যাসীয় কিন্তু এদের থেকে উৎপন্ন যৌগ পানি সাধারণ তাপমাত্রায় তরল three point two
 [!] Character '−' not found in the vocabulary. Discarding it.
যেমন− সাধারণ তাপমাত্রায় হাইড্রোজেন ও অক্সিজেন গ্যাসীয় কিন্তু এদের থেকে উৎপন্ন যৌগ পানি সাধারণ তাপমাত্রায় তরল three point two
 [!] Character '−' not found in the vocabulary. Discarding it.
যেমন− সাধারণ তাপমাত্রায় হাইড্রোজেন ও অক্সিজেন গ্যাসীয় কিন্তু এদের থেকে উৎপন্ন যৌগ পানি সাধারণ তাপমাত্রায় তরল three point two
 [!] Character '−' not found in the vocabulary. Discarding it.
 ! Run is kept in /content/drive/MyDrive/All/Personal/UBishops/ModelTraining/AAD/OutputForBangla/vits_vctk_old_mini_gdrive_ft-4000-March-13-2024_05+21AM-0000000
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/trainer/trainer.py", line 1833, in fit
    self._fit()
  File "/usr/local/lib/python3.10/dist-packages/trainer/trainer.py", line 1785, in _fit
    self.train_epoch()
  File "/usr/local/lib/python3.10/dist-packages/trainer/trainer.py", line 1503, in train_epoch
    for cur_step, batch in enumerate(self.train_loader):
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py", line 630, in __next__
    data = self._next_data()
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py", line 1345, in _next_data
    return self._process_data(data)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/data/dataloader.py", line 1371, in _process_data
    data.reraise()
  File "/usr/local/lib/python3.10/dist-packages/torch/_utils.py", line 694, in reraise
    raise exception
IndexError: Caught IndexError in DataLoader worker process 2.
Original Traceback (most recent call last):
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/data/_utils/worker.py", line 308, in _worker_loop
    data = fetcher.fetch(index)
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/data/_utils/fetch.py", line 51, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/usr/local/lib/python3.10/dist-packages/torch/utils/data/_utils/fetch.py", line 51, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/usr/local/lib/python3.10/dist-packages/TTS/tts/models/vits.py", line 280, in __getitem__
    return self.__getitem__(self.rescue_item_idx)
  File "/usr/local/lib/python3.10/dist-packages/TTS/tts/models/vits.py", line 263, in __getitem__
    item = self.samples[idx]
IndexError: list index out of range`

**Note**

After adding the missing characters the error remains the same. Thank you.

**Comment by mobassir94** [comment]
@adil-ahmed please check this code : https://github.com/mobassir94/comprehensive-bangla-tts/blob/main/bn_vits_tts/Bangla_phoneme_ViTS_trainer.ipynb
Using this code i successfully fine tuned vits for bangla. In my repository you'll find all codes that i used for Making bangla tts using vits,follow them carefully, good luck.

**Reply by adil-ahmed** [reply]
Hello brother,

Thank you for your response. I have tried your code and it works but unfortunately it discarded all the bengali characters. I have seached your repo where charactersConfig part was commented. I have opened a discussion, let's see if i can manage to get an answer. Thanks again.
